src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/ScriptStringPayloadGeneratorAdapterUIHandler.java

```diff
@@ -160,15 +186,99 @@ public ScriptStringPayloadGeneratorAdapterUIPanel(List<ScriptWrapper> scriptWrap
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(scriptLabel)
+                                    .addComponent(scriptComboBox))
+                    .addComponent(getPayloadsPreviewGenerateButton())
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(payloadsPreviewLabel)
+                                    .addComponent(payloadsPreviewScrollPane)));
+        }
+
+        private JButton getPayloadsPreviewGenerateButton() {
+            if (payloadsPreviewGenerateButton == null) {
+                payloadsPreviewGenerateButton = new JButton(PAYLOADS_PREVIEW_GENERATE_FIELD_LABEL);
+                payloadsPreviewGenerateButton.setEnabled(false);
+
+                payloadsPreviewGenerateButton.addActionListener(new ActionListener() {
 
-            layout.setVerticalGroup(layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
-                    .addComponent(scriptLabel)
-                    .addComponent(scriptComboBox));
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        updatePayloadsPreviewTextArea();
+                    }
+                });
+            }
+            return payloadsPreviewGenerateButton;
+        }
+
+        private JTextArea getPayloadsPreviewTextArea() {
+            if (payloadsPreviewTextArea == null) {
+                payloadsPreviewTextArea = new JTextArea(15, 10);
+                payloadsPreviewTextArea.setEditable(false);
+            }
+            return payloadsPreviewTextArea;
+        }
+
+        private void updatePayloadsPreviewTextArea() {
+            if (!validateScriptImpl()) {
+                return;
+            }
+            ScriptUIEntry scriptUIEntry = ((ScriptUIEntry) scriptComboBox.getSelectedItem());
+            ScriptWrapper scriptWrapper = scriptUIEntry.getScriptWrapper();
+            ScriptStringPayloadGenerator scriptPayloadGenerator = scriptUIEntry.getScriptPayloadGenerator();
+            StringBuilder contents = new StringBuilder();
+            try {
+                try (ResettableAutoCloseableIterator<StringPayload> itPayloads = new ScriptStringPayloadGeneratorAdapter(
+                        scriptWrapper,
+                        scriptPayloadGenerator).iterator()) {
+                    for (int i = 0; i < MAX_NUMBER_PAYLOADS_PREVIEW && itPayloads.hasNext(); i++) {
+                        if (contents.length() > 0) {
+                            contents.append('\n');
+                        }
+                        contents.append(itPayloads.next().getValue());
+                    }
+                    itPayloads.reset();
+                }
+                getPayloadsPreviewTextArea().setEnabled(true);
+            } catch (Exception ignore) {
```

Comment:
The same reason as before, script engine implementations, wrongly, throw internal exceptions.